tools/settings.js

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env node
+
+/**
+ * Outputs a JSON object representing the extension settings based on a settings
+ * file and the package environment.
+ */
+
+'use strict';
+
+const path = require('path');
+const gitDescribeSync = require('git-describe').gitDescribeSync;
+
+// Suppress (expected) EPIPE errors on STDOUT
```

Comment:
#### `test.js`:
```javascript
console.log('foo')
console.log('bar')
```
Then:
```bash
$ node test.js | head -1
foo
events.js:141
      throw er; // Unhandled 'error' event
      ^

Error: write EPIPE
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at WriteWrap.afterWrite (net.js:763:14)
```

Basically: Node decides that unlike every other major scripting language it's the programmer's job to catch the EPIPE that gets returned from the `write(2)` syscall when the output is truncated with SIGPIPE.